src/main/java/org/sagebionetworks/web/client/widget/team/controller/TeamEditModalWidget.java

```diff
@@ -159,18 +159,23 @@ private void show() {
     view.clear();
     view.configure(team);
     //SWC-6259
-    Boolean canPublicJoin = team.getCanPublicJoin();
+    //SWC-7000
+    Boolean canPublicJoin = team.getCanPublicJoin() == null
```

Comment:
Match logic to TeamPresenter.  canPublicJoin is true iff team.getCanPublicJoin().
Is locked down only if !canPublicJoin and team.getCanRequestMembership()===false